soh/soh/Enhancements/mods.cpp

```diff
@@ -1018,6 +1018,26 @@ void RegisterRandomizerSheikSpawn() {
     });
 }
 
+void RegisterHurtContainer() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnPlayerUpdate>([]() {
+        static uint16_t hurtInitialized = 0;
+        if (!CVarGetInteger("gHurtContainer", 0)) {
+            hurtInitialized = 0;
+            return;
+        }
+        // Max Health is 320 (20 Hearts)
+        
+        if (hurtInitialized == 0) {
+            uint16_t getHeartPieces = gSaveContext.sohStats.heartPieces / 4;
+            uint16_t getHeartContainers = gSaveContext.sohStats.heartContainers;
+            
+            gSaveContext.healthCapacity = 320 - ((getHeartPieces + getHeartContainers) * 16);
+
+            hurtInitialized = 1;
+        }
+    });
+}
```

Comment:
```suggestion
void UpdateHurtContainerModeState(bool newState) {
    static bool hurtEnabled = false;
    if (hurtEnabled == newState) {
        return;
    }
    
    hurtEnabled = newState;
    uint16_t getHeartPieces = gSaveContext.sohStats.heartPieces / 4;
    uint16_t getHeartContainers = gSaveContext.sohStats.heartContainers;
    
    if (hurtEnabled) {
        gSaveContext.healthCapacity = 320 - ((getHeartPieces + getHeartContainers) * 16);
    } else {
        gSaveContext.healthCapacity = (getHeartPieces + getHeartContainers) * 16;
    }
}

void RegisterHurtContainerModeHandler() {
    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnLoadGame>([](int32_t fileNum) {
        UpdateHurtContainerModeState(CVarGetInteger("gHurtContainer", 0));
    });
}
```

Reply:
Apologies for the ignorance, but wouldn't this make it so it would only check if the state change on File Load?

The newState also isn't being captured anywhere, doesn't look like it can be referenced directly for some reason.

Reply:
it checks on game load and on toggle (with the below change)

Reply:
updated Commit , needed to add `gSaveContext.healthCapacity = 48 - ((getHeartPieces + getHeartContainers) * 16);` to achieve the 3 heart minimum but overall it worked out.